FlowCrypt/Core/Models/KeyDetails.swift

```diff
@@ -40,6 +40,15 @@ extension KeyDetails {
     var pgpUserEmails: [String] {
         users.map { MCOAddress(nonEncodedRFC822String: $0).mailbox }
     }
+    
+    var isKeyUsuable: Bool {
+        var expired = false
+        let now = Date()
+        if let expiration = expiration, TimeInterval(expiration) < now.timeIntervalSince1970 {
```

Comment:
Maybe `expiration.toDate() < Date()` or similar? I noticed there's an extension for this

Then you don't need a `let now` . You also don't need `expired = false`

Finally, in another issue I just thought to look at `lastModified` as well as uids

Roughly this:

```swift
// revoked keys are not usable
guard !revoked else { return false }
// keys without lastModified don't have valid signatures on them - not usable
guard lastModified else { return false }
// keys without uids on them are not usable
guard users.isNotEmpty else { return false } 
// expired keys are not usable
if let expiration = expiration, expiration.toDate() < Date() { return false } 
// non-revoked keys, with lastModified and at least one user, that are not expired are usable
// gross simplification until https://github.com/FlowCrypt/flowcrypt-ios/issues/1546
return true
```